src/main/java/com/minecolonies/coremod/colony/managers/BuildingManager.java

```diff
@@ -628,19 +628,12 @@ public boolean hasGuardBuildingNear(final IBuilding building)
     @Override
     public void guardBuildingChangedAt(final IBuilding guardBuilding, final int newLevel)
     {
-        for (final IBuilding building : colony.getBuildingManager().getBuildings().values())
+        final int claimRadius = guardBuilding.getClaimRadius(Math.max(guardBuilding.getBuildingLevel(), newLevel)) * 16;
+        for (final IBuilding building : getBuildings().values())
         {
-            if (building.getPosition().getX() <= guardBuilding.getPosition().getX() + 16 * guardBuilding.getClaimRadius(newLevel)
-                  && building.getPosition().getZ() <= guardBuilding.getPosition().getZ() + 16 * guardBuilding.getClaimRadius(newLevel))
+            if (claimRadius > building.getID().distManhattan(guardBuilding.getID()))
```

Comment:
looks good now, just a small thing: you likely want to use a 2d manhattan distance so y levels dont affect it

Reply:
You might be right, but this is the same check being done in `hasGuardBuildingNear`.  So it's consistent as is.

And it would be a bit weird for a guard building at y = 100 to be successfully defending a building at y = 30...